laws/jvm/src/main/scala/algebra/laws/Platform.scala

```diff
@@ -7,28 +7,33 @@ import Prop.{False, Proof, Result}
 import scala.util.control.NonFatal
 
 private[laws] object Platform {
-
-  // Scala-js does not implement the Serializable interface, so the real test is for JVM only.
+  // Scala-js does not implement the Serializable interface, so the
+  // real test is for JVM only.
   @inline
-  def serializable[M](m: M): (String, Prop) = 
-    "serializable" -> Prop { _ =>
-      import java.io._
-      val baos = new ByteArrayOutputStream()
-      val oos = new ObjectOutputStream(baos)
-      var ois: ObjectInputStream = null
-      try {
-        oos.writeObject(m)
-        oos.close()
-        val bais = new ByteArrayInputStream(baos.toByteArray())
-        ois = new ObjectInputStream(bais)
-        val m2 = ois.readObject() // just ensure we can read it back
-        ois.close()
-        Result(status = Proof)
-      } catch { case NonFatal(t) =>
-        Result(status = Exception(t))
-      } finally {
-        oos.close()
-        if (ois != null) ois.close()
-      }
-    }
+  def serializable[M: IsSerializable](m: M): (String, Prop) =
+    "serializable" -> (implicitly[IsSerializable[M]] match {
+      case Is() =>
+        Prop(_ => Result(status = Proof))
+      case IsNot() =>
```

Comment:
Shouldn't we only be checking the serializability if it `Is` serializable instead of if it `IsNot`? :)